pages/api/blogposts/index.ts

```diff
@@ -1,14 +1,55 @@
 import { AuthenticatedRequest } from "@/src/auth/utils";
 import getBlogPostsInteractor from "@/src/blogposts/get-blogposts";
 import type { NextApiResponse } from "next";
+import prisma from "@/prisma";
+import {withAuth} from "@/src/auth/middleware";
 
-function handler(req: AuthenticatedRequest, res: NextApiResponse) {
+const addBlogPost = withAuth(async (req: AuthenticatedRequest, res: NextApiResponse) => {
+    const { title, description, content, tags } = req.body;
+    
+    // Check for missing fields
+    if (!title || !description || !content || !tags) {
+        return res.status(400).json({ error: 'Missing required fields' });
+    }
+
+    try {
+        const userId = Number(req.user.userId);
+
+        if (isNaN(userId)) {
+            console.error("Invalid user ID:", userId);
+            return res.status(400).json({ error: "Invalid user ID" });
+        }
+
+        const blogpost = await prisma.blogPost.create({
+            data: {
+                title,
+                description,
+                content,
+                postedAt: new Date(),
+                user: {
+                    connect: { id: userId }
+                },
+                tags: {
+                    create: tags.map((tag: string) => ({ name: tag })),
```

Comment:
Perhaps try using `upsert` here? Then we probably don't need to use the big block of code I used.